app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt

```diff
@@ -138,4 +145,21 @@ class SectionHolder(
     fun setBottomShadow(showShadow: Boolean) {
         formBinding.shadowBottom.visibility = if (showShadow) View.VISIBLE else View.GONE
     }
+
+    fun setLastSectionHeight(previousSectionIsOpened: Boolean) {
+        val params = formBinding.lastSectionDetails.layoutParams
+        val finalHeight = if (previousSectionIsOpened) {
+            48.dp
+        } else {
+            1.dp
+        }
+        ValueAnimator.ofInt(params.height, finalHeight).apply {
+            duration = 120
+            addUpdateListener {
+                params.height = it.animatedValue as Int
+                formBinding.lastSectionDetails.layoutParams = params
+            }
+            start()
```

Comment:
this animation should apply to all cases or only if previous section is Opened?

Reply:
All cases, open or closed.